zebra-chain/src/sapling/tests/tree.rs

```diff
@@ -109,12 +109,22 @@ fn incremental_roots_with_blocks_for_network(network: Network) -> Result<()> {
     );
 
     // Add note commitments in Block 1 to the tree
+    let mut appended_count = 0;
     for transaction in block1.transactions.iter() {
         for sapling_note_commitment in transaction.sapling_note_commitments() {
             tree.append(*sapling_note_commitment)
                 .expect("test vector is correct");
+            appended_count += 1;
         }
     }
+    // We also want to make sure that sapling_note_commitments() is returning
+    // the commitments in the right order. But this will only be actually tested
+    // if there are more than one note commitment in a block.
+    // In the test vectors this applies only for the block 1 in mainnet,
+    // so we make this explicit in this assert.
+    if network == Network::Mainnet {
+        assert!(appended_count > 1);
+    }
```

Comment:
Sapling isn't activated until hundreds of thousands of blocks after genesis.
How can there be any sapling note commitments in block 1?

Reply:
In this test, `block1` is Sapling activation height + 1

Reply:
Unless we want a name change for these test variables I think this is good to go!

Reply:
I'd like a quick fix to the comments and variable names to avoid confusion.

Everywhere else in the tests and in Zcash, "block 0" is the genesis block, and "block 1" is the block at height 1.

Reply:
Changed comments (and improved a bit of the structure) in 5ae1c531